jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/DEF.java

```diff
@@ -22,74 +22,110 @@
-                                                                          contentTypeRDFJSON,
-                                                                          contentTypeRDFThrift
-                                                                          );
-
-    public static final AcceptList quadsOffer         = AcceptList.create(contentTypeTriG,
-                                                                          contentTypeTriGAlt1,
-                                                                          contentTypeJSONLD,
-                                                                          contentTypeNQuads,
-                                                                          contentTypeNQuadsAlt1,
-                                                                          contentTypeTriX,
-                                                                          contentTypeTriXxml
-                                                                          );
+    // @formatter:off
+
+    // ---- Server configuration default media type
+
+    public static MediaType acceptRDFXML        = acceptRDFXMLDefault();
+    public static MediaType acceptNQuads        = acceptNQuadsDefault();
+    public static MediaType acceptResultSetXML  = acceptResultSetXMLDefault();
+    public static MediaType acceptJSON          = acceptJSONDefault();
+    public static MediaType acceptTurtle        = acceptTurtleDefault();
+
+    // ---- Server configuration offers for content negotiation
+
+    public static AcceptList jsonOffer          = jsonOfferDefault();
+
+    public static AcceptList constructOffer     = constructOfferDefault();
+
+    public static AcceptList rdfOffer           = rdfOfferDefault();
+
+    public static AcceptList quadsOffer         = quadsOfferDefault();
+
+    // Offer for SELECT
+    public static AcceptList rsOfferTable       = rsOfferTableDefault();
+
+    // Offer for ASK
+    public static AcceptList rsOfferBoolean     = rsOfferBooleanDefault();
+
+
+    // ---- Default configuration settings or when content negotiation does not provide a media type.
+
+    public static final MediaType acceptRDFXMLDefault()       { return MediaType.create(contentTypeRDFXML); }
+    public static final MediaType acceptNQuadsDefault()       { return MediaType.create(contentTypeNQuads); }
+    public static final MediaType acceptResultSetXMLDefault() { return MediaType.create(contentTypeResultsXML); }
+    public static final MediaType acceptJSONDefault()         { return MediaType.create(contentTypeJSON); }
+    public static final MediaType acceptTurtleDefault()       { return MediaType.create(contentTypeTurtle); }
+
+    // ---- Default offers for content negotiation.
+
+    public static final AcceptList jsonOfferDefault()         { return AcceptList.create(contentTypeJSON); }
+
+    public static final AcceptList constructOfferDefault()    { return AcceptList.create(contentTypeTurtle,
```

Comment:
Could the defaults be generated based on `RDFLanguages` or other suitable registry rather than hardcoding?

Then the OPs scenario in #2700 may just work without them needing any further code to modify the static fields

Obviously initialisation and registration order is going to be a thing so may need both approaches regardless, WDYT?

Reply:
> Could the defaults be generated based on `RDFLanguages` or other suitable registry rather than hardcoding?

1. The order in `Accept` and `offer` lists matters.
1. Maybe a user wants a server to offer a subset of what Jena can support.
1. `RDFLanguages` is a registry of constants. It does not guarantee a writer exists nor does it cover result sets.
1. Each offer type is context dependent. If it's a SELECT query, then `rsOfferTable` is used. If the result is a CONSTRUCT, `constructOffer`, if it is GSP, then it depends on graph vs dataset. It's not a choice. A registry would be an enum to offer with fixed choice of enum key. That's the same as asking for the right constant.